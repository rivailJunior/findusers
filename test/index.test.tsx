import React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import Index from '../src/pages/index'

afterEach(cleanup)

describe('Index Page', () => {
  test('Snapshot', () => {
    const index = renderer.create(<Index />)
    index.toJSON()
    expect(index).toMatchSnapshot()
  })

  test('Show user a button to login in github and some informations', () => {
    const { getByText } = render(<Index />)

    expect(
      getByText(/Olá Avaliador!/i)
    ).toBeInTheDocument()
    expect(
      getByText(
        /Para realizar algumas requisições precisamos dessa etapa inicial, o FindUser não ficará com nenhum dado seu./i
      )
    ).toBeInTheDocument()
    expect(getByText(/Login with Git/i)).toBeInTheDocument()
    expect(getByText(/Login with Git/i)).toHaveAttribute('href')
  })
})
