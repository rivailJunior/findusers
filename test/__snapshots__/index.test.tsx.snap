// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page Snapshot 1`] = `
<div
  className="container"
>
  <h2>
    Olá Avaliador!
  </h2>
  <h5>
    Para realizar algumas requisições precisamos dessa etapa inicial, o FindUser não ficará com nenhum dado seu.
  </h5>
  <ul>
    <li>
      Ao clicar no botão abaixo, você será será redirecionado ao login com o github.
    </li>
    <li>
      Após login, você será redirecionado a pagina inicial dessa avaliação
    </li>
  </ul>
  <a
    className="btn"
    href="https://github.com/login/oauth/authorize?client_id=undefined&redirect_uri:undefined&state=undefined&allow_singnup=true"
    type="button"
  >
    Login with Git
  </a>
</div>
`;
